liquidity-rewards/contracts/UniswapPoolRewards.sol

```diff
@@ -1,5 +1,193 @@
+        _notifyRewardAmount(reward, duration);
+        emit RewardPeriodStarted(rewardPeriod, reward, duration);
+    }
+
+    // Returns current timestamp if the rewards program has not finished yet, end time otherwise
+    function lastTimeRewardApplicable() public view returns (uint256) {
+        return Math.min(block.timestamp, periodFinish);
+    }
+
+    // Returns the amount of rewards that correspond to each staked token
+    function rewardPerToken() public view returns (uint256) {
+        if (totalSupply() == 0) {
+            return rewardPerTokenStored;
+        }
+        return
+            rewardPerTokenStored.add(
+                lastTimeRewardApplicable()
+                    .sub(lastUpdateTime)
+                    .mul(rewardRate)
+                    .mul(1e18)
+                    .div(totalSupply())
+            );
+    }
+
+    // Returns the amount that an account can claim
+    function earned(address account) public view returns (uint256) {
+        return
+            balanceOf(account)
+                .mul(rewardPerToken().sub(userRewardPerTokenPaid[account]))
+                .div(1e18)
+                .add(rewards[account]);
+    }
+
+    // stake visibility is public as overriding LPTokenWrapper's stake() function
+    function stake(uint256 amount) public override {
+        require(amount > 0, "Cannot stake 0");
+        require(address(lpToken) != address(0), "Liquidity Pool Token has not been set yet");
+
+        _updatePeriodFinish();
+        _updateAccountReward(msg.sender);
+
+        super.stake(amount);
+        periodTotalStaked = periodTotalStaked.add(amount);
+        emit Staked(msg.sender, rewardPeriod, amount);
+    }
+
+    function withdraw(uint256 amount) public override{
+        require(amount > 0, "Cannot withdraw 0");
+        require(address(lpToken) != address(0), "Liquidity Pool Token has not been set yet");
+
+        _updateAccountReward(msg.sender);
+
+        super.withdraw(amount);
+        periodTotalStaked = periodTotalStaked.sub(amount);
+        emit Withdrawn(msg.sender, rewardPeriod, amount);
+    }
+
+    // Shortcut to be able to unstake tokens and claim rewards in one transaction
+    function withdrawAndClaim() external {
+        withdraw(balanceOf(msg.sender));
+        claimReward();
+    }
+
+    function claimReward() public {
+        require(address(lpToken) != address(0), "Liquidity Pool Token has not been set yet");
+
+        _updatePeriodFinish();
+        _updateAccountReward(msg.sender);
+
+        uint256 reward = earned(msg.sender);
+
+        require(reward > 0, "Nothing to claim");
+
+        rewards[msg.sender] = 0;
+        totalRewardPaid = totalRewardPaid.add(reward);
+        periodRewardPaid = periodRewardPaid.add(reward);
+        elimuToken.transfer(msg.sender, reward);
+        emit RewardPaid(msg.sender, rewardPeriod, reward);
+    }
+
+
+    function _notifyRewardAmount(uint256 _reward, uint256 _duration) internal {
```

Comment:
@hhio618 What does "notify" mean in this context? Could you add a comment explaining this function?